Backend/src/main/java/com/example/backend/project/controller/ProjectController.java

```diff
@@ -34,225 +36,169 @@ public class ProjectController {
     public ResponseEntity updateProject(@RequestHeader String googleTokenEncoded, @RequestBody ProjectDTO projectDTO, @PathVariable Long id){
-        List<AUTHORITY> a = List.of(AUTHORITY.MODERATOR, AUTHORITY.ADMINISTRATOR);
-        String email = JwtVerifier.verifyAndReturnEmail(googleTokenEncoded);
-        if (email == null)
-            return new ResponseEntity("Token is missing or invalid", HttpStatus.UNAUTHORIZED);
-        if (userService.findByEmail(email) == null)
-            return new ResponseEntity("You don't have access to CDB", HttpStatus.UNAUTHORIZED);
-        if (!a.contains(userService.findByEmail(email).getAuthority()))
-            return new ResponseEntity("You don't have premission to this resource", HttpStatus.UNAUTHORIZED);
-
-        if (!projectService.existsById(id)) return new ResponseEntity("Project not found", HttpStatus.NOT_FOUND);
-        return new ResponseEntity(projectService.updateProject(projectDTO, id), HttpStatus.OK);
+        AppUser user = getUser(googleTokenEncoded);
+        try {
+            return new ResponseEntity(projectService.updateProject(projectDTO, id, user), HttpStatus.OK);
+        } catch (AuthenticationException e) {
+            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
+        } catch (UserNotFoundException e) {
+            return new ResponseEntity<>("User not found", HttpStatus.NOT_FOUND);
+        } catch (ProjectNotFoundException e){
+            return new ResponseEntity("Project not found", HttpStatus.NOT_FOUND);
+        }
     }
 
     @DeleteMapping("/{id}")
     public ResponseEntity deleteProjects(@RequestHeader String googleTokenEncoded, @PathVariable Long id){
-        List<AUTHORITY> a = List.of(AUTHORITY.MODERATOR, AUTHORITY.ADMINISTRATOR);
-        String email = JwtVerifier.verifyAndReturnEmail(googleTokenEncoded);
-        if (email == null)
-            return new ResponseEntity("Token is missing or invalid", HttpStatus.UNAUTHORIZED);
-        if (userService.findByEmail(email) == null)
-            return new ResponseEntity("You don't have access to CDB", HttpStatus.UNAUTHORIZED);
-        if (!a.contains(userService.findByEmail(email).getAuthority()))
-            return new ResponseEntity("You don't have premission to this resource", HttpStatus.UNAUTHORIZED);
-
-        if (!projectService.existsById(id)) return new ResponseEntity("Project not found", HttpStatus.NOT_FOUND);
-        projectService.deleteProject(id);
-        return new ResponseEntity("Project under id: " + id + " Successfully deleted", HttpStatus.OK);
+        AppUser user = getUser(googleTokenEncoded);
+        try {
+            projectService.deleteProject(id, user);
+            return new ResponseEntity(HttpStatus.OK);
+        } catch (AuthenticationException e) {
+            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
+        } catch (ProjectNotFoundException e){
+            return new ResponseEntity("Project not found", HttpStatus.NOT_FOUND);
+        }
     }
 
```

Comment:
svi ovi endpointovi s collaborations ne trebaju postojati: [endpoints](https://github.com/orgs/best-zagreb/projects/7/views/2?pane=issue&itemId=32159485)

Reply:
zasada nek ostanu tu jer nam je to trebalo prije, kasnije će to biti taskovi da se ti endpointovi zamijene